[WIP] Add installation with iscsi disk for agama

chcao opened the pull request:
Add installation with iscsi disk for agama.

- Related ticket: https://progress.opensuse.org/issues/176361
- Needles: n/a
- Verification run: TBD

jknphy commented on schedule/yam/agama_auto_iscsi.yaml:
```diff
@@ -0,0 +1,12 @@
+---
```
we try to use `unattended` instead of `auto` everywhere.

chcao replied:
thanks for review, I updated it.

jknphy commented on schedule/yam/agama_auto_iscsi.yaml:
```diff
@@ -0,0 +1,12 @@
+---
+name: agama_auto_iscsi
+description: >
+  Prepare url for agama.auto boot parameter, boot, perform auto-installation with agama on iscsi disk.
```
no need to explain how it works, just what does from functiona point of view, for example, next build will be `inst.auto` so then the code requires maintenance. Use `unattended` instead of `auto-installation`, 'Agama' instead of 'agama', should be `iscsi_ibft` and description should have that too.

chcao replied:
thanks for review, I updated it.

jknphy commented on tests/installation/validation/ibft.pm:
```diff
@@ -117,6 +118,9 @@ sub run {
 
     select_console 'root-console';
+    # Workaround for packages of lsscsi and hdparm are not pre-installed on sle16
+    zypper_call 'in lsscsi hdparm' if is_sle('>=16');
```
please check this again with latest build.

chcao replied:
the bugs were fixed, and removed the workaround, thanks.

jknphy commented:
Please update the test to search in the corresponding logs.

jknphy commented:
Kindly reminder of the missing things:
- Automation should fail in something related with ibft.
- Pull request should reflect ibft as well (not just iscsi).
- Changes in code are better explained in the first comment of your PR.
- Verification are provided for sles 16 and sles 15 when you change code affecting sles 15.

chcao commented:
> Kindly reminder of the missing things:
> 
> * Automation should fail in something related with ibft.

I changed the log to `/var/log/agama-installation/*` for sle16.

> * Pull request should reflect ibft as well (not just iscsi).

I updated the commit and the PR tittle.

> * Changes in code are better explained in the first comment of your PR.

Yes, I updated the first comment of this PR.

> * Verification are provided for sles 16 and sles 15 when you change code affecting sles 15.

Updated the vr links in the first comment, thanks.
